tagstudio/src/qt/widgets/video_player.py

```diff
@@ -0,0 +1,400 @@
+        # self.fullscreen_button.setMouseTracking(True)
+        # self.fullscreen_button.installEventFilter(self)
+        # self.scene().addWidget(self.fullscreen_button)
+        # self.fullscreen_button.resize(40, 40)
+        # self.fullscreen_button.move(self.fullscreen_button.size().width()/2, self.height() - self.fullscreen_button.size().height()/2)
+        # self.fullscreen_button.hide()
+
+        self.setContextMenuPolicy(Qt.ContextMenuPolicy.ActionsContextMenu)
+        self.opener = FileOpenerHelper(filepath=self.filepath)
+        autoplay_action = QAction("Autoplay", self)
+        autoplay_action.setCheckable(True)
+        if self.driver.settings.contains("autoplay_videos"):
+            if self.driver.settings.value("autoplay_videos") == "true":
+                autoplay_action.setChecked(True)
+            else:
+                autoplay_action.setChecked(False)
+        else:
+            autoplay_action.setChecked(True)
+        autoplay_action.triggered.connect(self.toggleAutoplay)
+        self.addAction(autoplay_action)
+        self.autoplay = autoplay_action
+
+        open_file_action = QAction("Open file", self)
+        open_file_action.triggered.connect(self.opener.open_file)
+        open_explorer_action = QAction("Open file in explorer", self)
+        open_explorer_action.triggered.connect(self.opener.open_explorer)
+        self.addAction(open_file_action)
+        self.addAction(open_explorer_action)
+
+    def close(self, *args, **kwargs) -> None:
+        self.player.stop()
+        super().close(*args, **kwargs)
+
+    def toggleAutoplay(self) -> None:
+        self.driver.settings.setValue("autoplay_videos", self.autoplay.isChecked())
+
+    def checkMediaStatus(self, media_status: QMediaPlayer.MediaStatus) -> None:
+        logging.info(media_status)
+        if media_status == QMediaPlayer.MediaStatus.EndOfMedia:
+            # Switches current video to with video at filepath. Reason for this is because Pyside6 is dumb and can't handle setting a new source and freezes.
+            # Even if I stop the player before switching, it breaks.
+            # On the plus side, this adds infinite looping for the video preview.
+            self.player.stop()
+            old_filename = self.player.source().fileName()
+            logging.info(f"Old filepath: {old_filename}")
+            self.player.setSource(QUrl().fromLocalFile(self.filepath))
+            # logging.info(f'Set source to {self.filepath}.')
+            # self.video_preview.setSize(self.resolution)
+            self.player.setPosition(0)
+            # logging.info(f'Set muted to true.')
+            extension = os.path.splitext(self.filepath)[1][1:].lower()
+            filename = os.path.basename(self.filepath)
+            if old_filename == filename:
+                self.player.play()
+            elif extension in VIDEO_TYPES:
+                # self.player.setVideoOutput(self.video_preview)
+                self.album_art.setOpacity(0.0)
```

Comment:
This line throws an error as of ec0d4cb7089a8d225ba2fe3573602bc240b68d77 since the `album_art` variable was removed